system/index.js

```diff
@@ -28,56 +41,46 @@ const userFn = getFunction(USER_FUNCTION_URI);
 
 module.exports = async (message) => {
   const payload = message.payload;
+
   // Remap headers to a standard JS object
   const headers = message.headers.toRiffHeaders();
-  Object.keys(headers).map((key) => {headers[key] = message.headers.getValue(key)});
-  if (DEBUG) {
-    console.log('==System Function Start==');
-    console.log(`HEADERS: ${JSON.stringify(headers)}`);
-    console.log(`PAYLOAD: ${JSON.stringify(payload)}`);
-    console.log(`MIDDLEWARE_FUNCTION_URI: ${MIDDLEWARE_FUNCTION_URI}`);
-    console.log('==Middleware Function(s) Start==');
-  }
+  Object.keys(headers).map((key) => { headers[key] = message.headers.getValue(key) });
+
+  const logLevel = DEBUG ? SF.LoggerLevel.DEBUG : SF.LoggerLevel.INFO;
+  const requestId = headers['ce-id'] || headers['x-request-id'];
+  const logger = await createLogger(logLevel, requestId);
 
   const state = {};
-  let middlewareResult = [payload];
+  let middlewareResult = [payload, logger];
```

Comment:
Short answer: If we do have multiple middlewares, they can choose to forward the logger on or omit to to control the signature. Our default signature is now `(payload, logger)` but any middleware can change that to whatever they want. While we could pass the logger in the `input`, that was sort of intended to be a way to access the original, unmodified payload + headers. This was so that middleware could potentially act upon a payload which has potentially been unwrapped/parsed into something else by other middleware. 

Longer option: This is one part of the middleware design I don't like right now. One middleware can change the signature and another middleware can just undo everything it just did. Or another middleware can expect the arguments to the function to be something and they turn out to have been modified by middleware before. I think this is part of the function signature discussion, which _could_ eliminate or change what middleware means entirely.